staging/src/k8s.io/kubectl/pkg/cmd/completion/completion.go

```diff
@@ -307,7 +307,7 @@ BASH_COMPLETION_EOF
 }
 
 __kubectl_bash_source <(__kubectl_convert_bash_to_zsh)
-_complete kubectl 2>/dev/null
+_complete kubectl 2>/dev/null || true
```

Comment:
masking failure here does not make sense to me. I don't have the context to know what zsh requires, but if this is not needed, why not omit it entirely?

Reply:
Thanks for the feedback @liggitt 

Yeah, my first instinct was to completely remove it because it doesn't seem to be working for anyone based on the comments in the [github issue](https://github.com/kubernetes/kubectl/issues/125). Removing it worked for me, but I was concerned because I couldn't determine what it was *supposed* to be doing in case it had some purpose I was unaware of.

I found the [commit where the line was added](https://github.com/kubernetes/kubernetes/commit/70b942d6638e99484fe9eb11e878d7300dd8d9ae), but still don't get a feel for why it was needed (or why stderr was being ignored).  I investigated zsh, bash, and cobra repos and couldn't find any mention of an implementation of `_complete` or `_bash_comp`

I guess I could go either way.  

I don't see adding `|| true` as being much worse than how it already is with `2> /dev/null`.

But on the other hand, adding `|| true` perpetuates some tech debt in that someday, someone else will come along and also have the same discussion about why it is like that and whether it can or can't be removed.

Do you have any additional context that would be helpful in deciding how to close out the issue?

If I revert back to my original commit of removing line 310 and also removing the alias on line 161, would you approve that change?

Reply:
Hi @cblecker .  Can you shed some light on why `_complete` is needed?  It has [been failing and causing the exit code to be 127](https://github.com/kubernetes/kubectl/issues/125) because _bash_comp (which it is aliased to) is not found.

I know it has been a couple of years, but the line in question is below from [this Aug 12, 2017 commit](https://github.com/kubernetes/kubernetes/commit/70b942d6638e99484fe9eb11e878d7300dd8d9ae#diff-339bcec9f0f8e5768e0f98ad47c40361R308) ...
```
_complete kubectl 2>/dev/null
```

I think this line can just be removed since it is failing and completion still seems to work, but wanted to circle back with you and make sure there is not some other specific reason it is needed since it looks like you were the one who added it.

Reply:
Seconded. We should get rid of `_complete` because it is aliased to `_bash_comp`, which is *undefined* (https://github.com/kubernetes/kubectl/issues/125#issuecomment-392754413).

Reply:
@antoineco @liggitt ptal